### PR TITLE
quincy: doc/start: remove mention of Centos 8 support

### PR DESCRIPTION
**Pull request by zdover23**
Remove information so that it is clear that Centos 8 will no longer be supported by Ceph.

Signed-off-by: Zac Dover <zac.dover@proton.me>
(cherry picked from commit 6b149e4f6ab2ac716e73f42b0d6c67482a768da9)

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [x] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [x] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

File: doc/start/os-recommendations.rst [hunk 1]
```diff
@@ -59,6 +59,36 @@ distribution that includes a supported kernel and supported system startup
 framework, for example ``sysvinit`` or ``systemd``. Ceph is sometimes ported to
 non-Linux systems but these are not supported by the core Ceph effort.
 
+
++---------------+-----------------+------------------+------------------+
+|               | Quincy (17.2.z) | Pacific (16.2.z) | Octopus (15.2.z) |
++===============+=================+==================+==================+
```
Comment (idryomov) [inline doc/start/os-recommendations.rst:65]: The backport is adding a second/duplicate table -- there already is one in the quincy branch, just formatted differently, see lines 43-55 in this file.  I'd have expected the new note to be added there.

Comment (zdover23) [reply doc/start/os-recommendations.rst:65]: I'm in the middle of the process of repairing this. This backport was generate by a homemade script that I wrote a couple of years ago, and in cases like this where the information in the release branches diverges, the commits have to be hand-tooled after their generation.

Give me half an hour to get this into shape.

File: doc/start/os-recommendations.rst [hunk 2]
```diff
@@ -43,7 +43,7 @@ package (i.e., sysvinit, systemd).
 +--------------+--------+------------------------+--------------------------------+-------------------+-----------------+
 | Release Name | Tag    | CentOS                 | Ubuntu                         | OpenSUSE :sup:`C` | Debian :sup:`C` |
 +==============+========+========================+================================+===================+=================+
-| Quincy       | 17.2.z | 8 :sup:`A`             | 20.04 :sup:`A`                 | 15.3              | 11              |
+| Quincy       | 17.2.z | 8 :sup:`A` 9 :sup:`1`  | 20.04 :sup:`A`                 | 15.3              | 11              |
```
Comment (idryomov) [inline doc/start/os-recommendations.rst:46]: It looks like centos 8 is still mentioned?  The way I'd interpret
```
8 :sup:`A` 9 :sup:`1`
```
is "8 has been and continues to be A and something is added for 9".  That is not a message that we trying to send.

File: doc/start/os-recommendations.rst [hunk 3]
```diff
@@ -43,7 +43,7 @@ package (i.e., sysvinit, systemd).
 +--------------+--------+------------------------+--------------------------------+-------------------+-----------------+
 | Release Name | Tag    | CentOS                 | Ubuntu                         | OpenSUSE :sup:`C` | Debian :sup:`C` |
 +==============+========+========================+================================+===================+=================+
-| Quincy       | 17.2.z | 8 :sup:`A`             | 20.04 :sup:`A`                 | 15.3              | 11              |
+| Quincy       | 17.2.z | 9 :sup:`1`             | 20.04 :sup:`A`                 | 15.3              | 11              |
```
Comment (idryomov) [inline doc/start/os-recommendations.rst:46]: This doesn't make it clear what status/letter is applied to 9 IMO.

In main, `1` is a footnote on `A` -- it's not standalone.  Here, `1` appears to be just standing on itself.

Comment (zdover23) [reply doc/start/os-recommendations.rst]: I have made it so that the text has a `9` followed by a superscript `A` followed by a superscript `1`.

Comment (idryomov) [reply doc/start/os-recommendations.rst]: LGTM, although I don't know how it renders because Read the Docs check remains broken in quincy.

**Review by idryomov — APPROVED** [review]

**Comment by idryomov** [comment]
@zdover23 Someone pointed to me that https://docs.ceph.com/en/quincy/start/os-recommendations/ still shows the old table despite this PR being merged two weeks ago.  I checked and the same seems to be the case for reef: https://docs.ceph.com/en/reef/start/os-recommendations/.

Is this known?  If so, is this something that the Python-involving backports that you are working on are expected to address?

**Comment by zdover23** [comment]
> @zdover23 Someone pointed out to me that https://docs.ceph.com/en/quincy/start/os-recommendations/ still shows the old table despite this PR being merged two weeks ago. I checked and the same seems to be the case for reef: https://docs.ceph.com/en/reef/start/os-recommendations/.
> 
> Is this known? If so, is this something that the Python-involving backports that you are working on are expected to address?

Yes.

I have been working on this since the 13th of June. 

The three PRs that need to be backported to address this are:
https://github.com/ceph/ceph/pull/57685
https://github.com/ceph/ceph/pull/57650
https://github.com/ceph/ceph/pull/57700

If these are finally backported and the documentation is still not updating, then we will need to look more closely at https://tracker.ceph.com/issues/66201.

I am not an expert in this area, which I take to be the breaking of Python dependencies.

**Comment by zdover23** [comment]
> > @zdover23 Someone pointed out to me that https://docs.ceph.com/en/quincy/start/os-recommendations/ still shows the old table despite this PR being merged two weeks ago. I checked and the same seems to be the case for reef: https://docs.ceph.com/en/reef/start/os-recommendations/.
> > Is this known? If so, is this something that the Python-involving backports that you are working on are expected to address?
> 
> Yes.
> 
> I have been working on this since the 13th of June.
> 
> The three PRs that need to be backported to address this are: #57685 #57650 #57700
> 
> If these are finally backported and the documentation is still not updating, then we will need to look more closely at https://tracker.ceph.com/issues/66201.
> 
> I am not an expert in this area, which I take to be the breaking of Python dependencies.

@idryomov,

Here is a little more context.

This this the Quincy backport of PR#57685: https://github.com/ceph/ceph/pull/58637. It is not possible to do a simple cherry-pick of the change made in that PR because the functions that are in 57685 are moved to files that did not yet exist in Quincy. I have tried to alter the functions changed by 57685, but I run into this error:

```
zdover@NUC8i7BEH:~/Documents/ceph/src/cephadm$ tox -e flake8
flake8 installed: flake8==5.0.4,flake8-quotes==3.4.0,mccabe==0.7.0,pycodestyle==2.9.1,pyflakes==2.5.0
flake8 run-test-pre: PYTHONHASHSEED='208733664'
flake8 run-test: commands[0] | flake8 --config=tox.ini cephadm
cephadm:901:13: F841 local variable 'version' is assigned to but never used
1     F841 local variable 'version' is assigned to but never used
ERROR: InvocationError for command /home/zdover/Documents/ceph/src/cephadm/.tox/flake8/bin/flake8 --config=tox.ini cephadm (exited with code 1)
________________________________________________ summary _________________________________________________
ERROR:   flake8: commands failed
```
Here is the problem:

```
cephadm:901:13: F841 local variable 'version' is assigned to but never used
```

I am very reluctant to start making alterations to Python files in an architecture that I did not create and that I do not claim to understand. If anyone knows how to get this working, I'll gratefully accept the help.

**Comment by idryomov** [comment]
@zdover23 I did a mechanical backport (no testing!) locally, if you want me to I can push to your branch.

**Comment by zdover23** [comment]
> @zdover23 I did a mechanical backport (no testing!) locally, if you want me to I can push to your branch.

@idryomov, I say go for it.

**Comment by idryomov** [comment]
> @idryomov, I say go for it.

Done.

**Pull request merged**